### PR TITLE
raise error if fit has a 0 or inf

### PR DESCRIPTION
**Pull request by mjburton11**
@bqpd and @whoburg ready for review.  Sometimes the resulting fit contains a 0 or inf value.  I think it should return an error if this happens.

### CODE REVIEW

File: gpfit/fit.py [hunk 1]
```diff
@@ -77,36 +77,48 @@ def fit(xdata, ydata, K, ftype="ISMA"):
     A = params[[i for i in range(K*(d+1)) if i % (d + 1) != 0]]
     B = params[[i for i in range(K*(d+1)) if i % (d + 1) == 0]]
 
+    cs = []
+    exps = []
     if ftype == "ISMA":
         alpha = 1./params[range(-K, 0)]
+        for k in range(K):
+            cs.append(exp(alpha[k]*B[K]))
```
Comment (mjburton11) [inline gpfit/fit.py]: @bqpd I reverted back to this because I was having problems with higher dimensions using lhs, and rhs.

**Comment by bqpd** [comment]
Looks great! Any clear way to reduce the number of lines of code?

**Comment by mjburton11** [comment]
I tried reducing it by creating cs and exps from lhs and rhs but ran into some issues with higher dimensions.  Any ideas?

**Comment by bqpd** [comment]
If you use `.min()` it'll search across any number of dimensions.

**Comment by mjburton11** [comment]
@bqpd @galbramc why did it only fail on reynolds?

**Comment by bqpd** [comment]
It had an extra large RMS error? Non-determinism? different versions of
numpy interpreting the same SEED differently? Try running it again.

(I am not a huge fan of this unit-testing setup)

**Comment by mjburton11** [comment]
Test this please

**Comment by mjburton11** [comment]
Test this please

**Comment by mjburton11** [comment]
@bqpd can we set a seed so that it will be more deterministic or something?

**Comment by bqpd** [comment]
I think we do! This seems fairly consistent though, so maybe it's a real bug?

**Comment by mjburton11** [comment]
How can I figure out the bug? I don't think I changed the way the rms is calculated so I don't know why it would be giving different answers.

**Comment by bqpd** [comment]
Ack, I'll take a look tomorrow. These tests were by far the most frustrating part of refactoring `fit`

**Comment by bqpd** [comment]
it passes on my computer, frustrating.

**Comment by mjburton11** [comment]
wow, very frustrating...

**Comment by bqpd** [comment]
Yeah, my best guess is that a different numpy version is interpreting the fixed seed differently. I increased tolerance a bit, it's still a darn low error.

**Comment by mjburton11** [comment]
OK well I'm going to go ahead and merge this then

**Comment by bqpd** [comment]
woot!

**Pull request merged**